Lib/pdb.py

```diff
@@ -2677,7 +2678,7 @@ def _send(self, **kwargs):
         try:
             self._sockfile.write(json_payload.encode() + b"\n")
             self._sockfile.flush()
-        except OSError:
+        except (OSError, ValueError):
```

Comment:
What's the new `ValueError` about?

Reply:
I hit it while testing something unrelated. Seems like we can either get an `OSError` if the socket has been disconnected but `.close()` hasn't been called on the sockfile yet, or a `ValueError` if `.close()` has been called.

I think that case happens if we try to send a message after `detach()` has been called, though I'm not sure how exactly I triggered it. Maybe sending a `KeyboardInterrupt` that arrived after a `q` was processed?